src/I18nProvider.tsx

```diff
@@ -42,6 +42,11 @@ function plural(
 ): string {
   if (!query || typeof query.count !== 'number') return key
 
+  const nestedKey = `${key}.${pluralRules.select(query.count)}`
+  if (getDicValue(dic, nestedKey) !== undefined) {
+    return nestedKey
+  }
+
   const numKey = `${key}_${query.count}`
```

Comment:
Do you think that it would make sense to also allow this nested format?

Reply:
Hm do you mean something like this?

```
{
  "cart-message": {
     "cart-message_0": "",
     ...
}
```

If so, I think it would actually already work — i'm not sure how useful it is though — because this is only checking if the child key has one of the plural rules as a direct child. And if it does not, then the old rules will work the same.

Reply:
I was thinking with:

```
{
  "cart-message": {
     "0": "",
     ...
}
```

Although the supported plurals are only 6, we also support an exact match, so perhaps it is also interesting to allow nesting.

```
{
  "cart-message": {
     "42: "Exact match when {{count}} === 42",
     "one": "...", 
     "other": ".."
  } 
}
```

Reply:
Ah okay I got you now! Yeah so for example Smartling, the reason nested support is needed, only supports `one` and `other` it doesn't even support `zero` let alone the 3 others. So I am not sure how useful it would be...I am more then happy to do it though if you think we should.

Reply:
Sounds good! Thanks for your time 🙏

Reply:
Hey @aralroca just to confirm, should I add this feature or not? 😄

Reply:
yes @josephfarina, it would be great! 😊

Reply:
Done @aralroca!